.buildkite/x-pack/elastic-agent/scripts/steps/build-agent-core.sh

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+echo "+++ Build Agent artifacts"
+
+SNAPSHOT=""
+
+BEAT_VERSION=$(grep -oE '[0-9]+\.[0-9]+\.[0-9]+(\-[a-zA-Z]+[0-9]+)?' "libbeat/version/version.go")
+BEAT_VERSION_FULL=$BEAT_VERSION
+if [ "$DRA_WORKFLOW" == "snapshot" ]; then
+    SNAPSHOT="true"
+    BEAT_VERSION_FULL="${BEAT_VERSION}-SNAPSHOT"
+fi
+
+SNAPSHOT=$SNAPSHOT mage -d x-pack/elastic-agent packageAgentCore
+chmod -R 777 x-pack/elastic-agent/build/distributions
```

Comment:
Not sure why we are applying the 777 permissions recursively here, the archives we have do not need to be executable, maybe we want to split this between directory and files permissions:

```suggestion
chmod 777 x-pack/elastic-agent/build/distributions
chmod 664 x-pack/elastic-agent/build/distributions/*
```